packages/client/test/put.spec.js

```diff
@@ -0,0 +1,96 @@
+import * as assert from 'uvu/assert'
+import randomBytes from 'randombytes'
+import { Web3Storage } from 'web3.storage'
+import { Web3File } from 'web3-file'
+
+describe('put', () => {
+  const { AUTH_TOKEN, API_PORT } = process.env
+  const token = AUTH_TOKEN || 'good'
+  const endpoint = new URL(API_PORT ? `http://localhost:${API_PORT}` : '')
+
+  it('errors without token', async () => {
+    // @ts-ignore
+    const client = new Web3Storage({ endpoint })
+    const files = prepareFiles()
+    try {
+      await client.put(files)
+      assert.unreachable('should have thrown')
+    } catch (err) {
+      assert.is(err.message, 'missing token')
+    }
+  })
+
+  it('errors without content', async () => {
+    const client = new Web3Storage({ endpoint, token })
+    try {
+      await client.put([])
+      assert.unreachable('should have thrown')
+    } catch (err) {
+      assert.match(err.message, /input could not be parsed correctly/)
+    }
+  })
+
+  it('erros with a File that will not be parsed by the Cluster', async function () {
+    this.timeout(35e10) // This needs to happen because of retry...
```

Comment:
Perhaps add timeout and retry as configurable options to put?

Reply:
Added to options as suggested, with default value from constant